src/components/BulkActions/tests/BulkActions.test.tsx

```diff
@@ -283,6 +296,96 @@ describe('<BulkActions />', () => {
         expect(bulkActionButtons).toHaveLength(3);
         warnSpy.mockRestore();
       });
+
+      it('renders a BulkActionMenu when promotedActions are menus', () => {
+        const warnSpy = jest.spyOn(console, 'warn');
+        warnSpy.mockImplementation(() => {});
+        const bulkActionProps: Props = {
+          bulkActions: [],
+          promotedActions: [
+            {
+              title: 'button1',
+              actions: [
+                {
+                  content: 'action1',
+                },
+                {
+                  content: 'action2',
+                },
+              ],
+            },
+            {
+              title: 'button2',
+              actions: [
+                {
+                  content: 'action1',
+                },
+                {
+                  content: 'action2',
+                },
+              ],
+            },
+            {
+              content: 'button 2',
+            },
+            {
+              content: 'button 3',
+            },
+          ],
+          paginatedSelectAllText: 'paginated select all text string',
+          selected: false,
+          accessibilityLabel: 'test-aria-label',
+          label: 'Test-Label',
+          disabled: false,
+        };
+        const bulkActions = mountWithAppProvider(
+          <BulkActions {...bulkActionProps} />,
+        );
+        const bulkActionButtons = bulkActions.find(BulkActionButton);
+        expect(bulkActionButtons).toHaveLength(4);
+        const bulkActionMenus = bulkActions.find(BulkActionMenu);
+        expect(bulkActionMenus).toHaveLength(2);
+        warnSpy.mockRestore();
```

Comment:
I know this pattern was already used in this file, but should we move this + spyOn to a beforeEach/afterEach?